### ISSUE TITLE
Take stream as input

### ISSUE THREAD

**Issue by zhangjinzhou** [naturalatlas/node-gdal#241]
Is it possible to take stream as input?

**Comment by MidasXIV** [comment]
Hey, I have been trying to use an S3 file path to a .TIF file but i don't think it's supported. 
Have you found out any way to overcome this ?

**Comment by zhangjinzhou** [comment]
No. I researched, and it seems spatial files rarely support stream because of file structure.

**Comment by jdesboeufs** [comment]
Many formats support streaming: GeoJSON, GML, CSV, GPX, KML…
Currently `node-gdal` seems not to support stream in. You had to write your data to the filesystem, et read from it.
But you can stream out easily.

**Comment by timmycam** [comment]
@MidasXIV Did you figure out or come across a solution for reading from s3? I'm sure this is common process! There is a solution for the gdal lib [here](https://gis.stackexchange.com/questions/201831/how-to-efficiently-access-files-with-gdal-from-an-s3-bucket-using-vsis3/202887) but it throws an error from the node bindings.
